composer/trainer/trainer.py

```diff
@@ -2983,8 +2983,17 @@ def _eval_loop(
                                                 outputs[k] = v.cpu()
                                             else:
                                                 outputs[k] = v
-                                    else:
+                                    elif isinstance(self.state.outputs, Sequence):
+                                        outputs = []
+                                        for v in self.state.outputs:
+                                            if isinstance(v, torch.Tensor):
+                                                outputs.append(v.cpu())
+                                            else:
+                                                outputs.append(v)
+                                    elif isinstance(self.state.outputs, torch.Tensor):
                                         outputs = self.state.outputs.cpu()
+                                    else:
+                                        raise ValueError('Unsupported type for state.outputs in trainer eval loop.')
```

Comment:
To be a bit safer, I think we don't want to error here. Since the output of `eval_forward` is `Any`, the user could actually be returning a custom object (that could in theory have `.cpu()` defined on it. We don't want to break any existing workflows doing something like this. So I would leave the fallback case of `outputs = self.state.outputs.cpu()`. Does this make sense to you?

Reply:
Made the change as you suggested, thanks!